libraries/plugins/elasticsearch/elasticsearch_plugin.cpp

```diff
@@ -0,0 +1,366 @@
+
+      void update_account_histories( const signed_block& b );
+
+      graphene::chain::database& database()
+      {
+         return _self.database();
+      }
+
+      elasticsearch_plugin& _self;
+      primary_index< simple_index< operation_history_object > >* _oho_index;
+
+      std::string _elasticsearch_node_url = "http://localhost:9200/";
+      uint32_t _elasticsearch_bulk_replay = 10000;
+      uint32_t _elasticsearch_bulk_sync = 100;
+      bool _elasticsearch_logs = true;
+      bool _elasticsearch_visitor = false;
+   private:
+      void add_elasticsearch( const account_id_type account_id, const operation_history_id_type op_id );
+
+};
+
+elasticsearch_plugin_impl::~elasticsearch_plugin_impl()
+{
+   return;
+}
+
+void elasticsearch_plugin_impl::update_account_histories( const signed_block& b )
+{
+   graphene::chain::database& db = database();
+   const vector<optional< operation_history_object > >& hist = db.get_applied_operations();
+   for( const optional< operation_history_object >& o_op : hist ) {
+      optional <operation_history_object> oho;
+
+      auto create_oho = [&]() {
+         return optional<operation_history_object>(
+               db.create<operation_history_object>([&](operation_history_object &h) {
+                  if (o_op.valid())
+                     h = *o_op;
+               }));
+      };
+
+      if( !o_op.valid() ) {
+         _oho_index->use_next_id();
+         continue;
+      }
+      oho = create_oho();
+
+      const operation_history_object& op = *o_op;
+
+      // get the set of accounts this operation applies to
+      flat_set<account_id_type> impacted;
+      vector<authority> other;
+      operation_get_required_authorities( op.op, impacted, impacted, other ); // fee_payer is added here
+
+      if( op.op.which() == operation::tag< account_create_operation >::value )
+         impacted.insert( op.result.get<object_id_type>() );
+      else
+         graphene::app::operation_get_impacted_accounts( op.op, impacted );
+
+      for( auto& a : other )
+         for( auto& item : a.account_auths )
+            impacted.insert( item.first );
+
+      for( auto& account_id : impacted )
+      {
+         add_elasticsearch( account_id, oho->id );
+      }
+   }
+}
+
+void elasticsearch_plugin_impl::add_elasticsearch( const account_id_type account_id, const operation_history_id_type op_id )
+{
+   graphene::chain::database& db = database();
+   const auto &stats_obj = account_id(db).statistics(db);
+
+   // add new entry
+   const auto &ath = db.create<account_transaction_history_object>([&](account_transaction_history_object &obj) {
+      obj.operation_id = op_id;
+      obj.account = account_id;
+      obj.sequence = stats_obj.total_ops + 1;
+      obj.next = stats_obj.most_recent_op;
+   });
+
+   // keep stats growing as no op will be removed
+   db.modify(stats_obj, [&](account_statistics_object &obj) {
+      obj.most_recent_op = ath.id;
+      obj.total_ops = ath.sequence;
+   });
```

Comment:
This will lead to double increments when both the classic account history plugin and the ES plugin are enabled.

Reply:
i think we should not allow to run this 2 plugins together as that can bring troubles not only with the double increments in stats but also as ES is deleting stuff from object indexes, account_history could be still using that data and similar problems.

i made this control here, not sure if it is the best place: https://github.com/bitshares/bitshares-core/pull/444/commits/a6b2177a184b03cf42e28b3c357350bf0c4c29e2